src/fondant/pipeline/pipeline.py

```diff
@@ -362,16 +376,41 @@ def read(
                 msg,
             )
 
-        operation = ComponentOp(
-            name_or_path,
-            produces=produces,
-            arguments=arguments,
-            input_partition_rows=input_partition_rows,
-            resources=resources,
-            cache=cache,
-            cluster_type=cluster_type,
-            client_kwargs=client_kwargs,
-        )
+        if isinstance(ref, (str, Path)):
+            operation = ComponentOp.from_component_yaml(ref)
+        else:
+            if not ref.is_python_component():
+                err = """Only Python components are currently supported.
+                Make sure your class is decorated with the @python_component decorator."""
+                raise ValueError(err)
+            name = ref().__class__.__name__
```

Comment:
need to instantiate the class to get the name

Reply:
Isn't `ref == ref().__class__`?
I think `ref.__name__` should work as well.

Reply:
Or is this because of the decorator?